src/main/tables.ts

```diff
@@ -170,41 +157,41 @@ export async function importCSV(
   aidColumns: string[],
   signal: AbortSignal,
 ): Promise<{ aborted: boolean }> {
-  const client = new pg.Client(connectionConfig);
-  await client.connect();
-
-  const columnsSQL = columns.map((column) => `"${column.name}" ${column.type}`).join(', ');
-  console.info(`Table schema: ${columnsSQL}.`);
-
-  const fieldProcessors = columns.map((column) => getFieldProcessor(column.type, parseOptions));
-
   try {
-    await client.query(`BEGIN`);
-    // TODO: should we worry about (accidental) SQL-injection here?
-    await client.query(`DROP TABLE IF EXISTS "${tableName}"`);
-    await client.query(`CREATE TABLE "${tableName}" (${columnsSQL})`);
-    await copyRowsFromCsvFile(client, signal, fileName, fieldProcessors, tableName);
-    if (aidColumns.length > 0) {
-      if (aidColumns.includes(ROW_INDEX_COLUMN)) {
-        await client.query(`ALTER TABLE "${tableName}" ADD COLUMN IF NOT EXISTS "${ROW_INDEX_COLUMN}" SERIAL`);
+    await sql.begin(async (sql) => {
+      await sql`DROP TABLE IF EXISTS ${sql(tableName)}`;
+
+      const columnsSQL: SqlFragment[] = columns.map((column, i) =>
```

Comment:
does this make sql-injection (as in the comment) fully protected against? if so, you can mark the #13 as fixed by this PR

Reply:
Yes, no more injections, except the `Hack:` part which is a workaround...